src/table/virtualized-table/hooks/use-header-and-totals-height.ts

```diff
@@ -0,0 +1,75 @@
+import { useEffect } from 'react';
+import { VariableSizeList } from 'react-window';
+import { Column, PageInfo, TotalsPosition } from '../../../types';
+import { TableContext, useContextSelector } from '../../context';
+import { COMMON_CELL_STYLING } from '../../styling-defaults';
+import { GeneratedStyling } from '../../types';
+import { fontSizeToRowHeight } from '../../utils/styling-utils';
+import { MIN_BODY_ROW_HEIGHT, PADDING_TOP_BOTTOM } from '../constants';
+import useDynamicRowHeight from './use-dynamic-row-height';
+
+interface UseHeaderAndTotalsHeightProps {
+  columns: Column[];
+  columnWidths: number[];
+  pageInfo: PageInfo;
+  totalsPosition: TotalsPosition;
+  headerRef: React.RefObject<VariableSizeList<any>>;
+  totalsRef: React.RefObject<VariableSizeList<any>>;
+}
+
+const getBodyRowHeight = ({ fontSize = COMMON_CELL_STYLING.fontSize }: GeneratedStyling) =>
+  Math.max(MIN_BODY_ROW_HEIGHT, fontSizeToRowHeight(fontSize));
+
+const useHeaderAndTotalsHeight = ({
```

Comment:
Since it also includes the body height. Better name it something more generic? Ex. "useHeights"

Reply:
ah yes, I added the bodyHeight after naming it. I'll change it